pkg/webhooks/resource/vpol/handler.go

```diff
@@ -62,16 +71,63 @@ func (h *handler) Validate(ctx context.Context, logger logr.Logger, admissionReq
 	var group wait.Group
 	defer group.Wait()
 	group.Start(func() {
-		if validation.NeedsReports(admissionRequest, *response.Resource, h.admissionReports, h.reportConfig) {
-			err := h.admissionReport(ctx, request, response)
-			if err != nil {
-				logger.Error(err, "failed to create report")
-			}
-		}
+		h.audit(ctx, logger, admissionRequest, request, response)
 	})
 	return h.admissionResponse(request, response)
 }
 
+func (h *handler) audit(ctx context.Context, logger logr.Logger, admissionRequest handlers.AdmissionRequest, request vpolengine.EngineRequest, response vpolengine.EngineResponse) {
+	blocked := false
+	for _, p := range response.Policies {
+		if p.Actions.Has(admissionregistrationv1.Deny) {
+			blocked = true
+			break
+		}
+	}
+
+	enginResponses := make([]engineapi.EngineResponse, 0, len(response.Policies))
```

Comment:
Looks like a small typo 😅:

```suggestion
	engineResponses := make([]engineapi.EngineResponse, 0, len(response.Policies))
```